logs_app/operations.py

```diff
@@ -0,0 +1,31 @@
+import pandas as pd
+
+
+# Return most used IP address by using pandas mode(), then selecting element [0], which is the address
+def mf_ip(ip_df: pd.DataFrame) -> str:
+    series_mf = ip_df["Client IP address"].mode()
+    return series_mf[0]
+
+
+# Return least used IP address by pandas idxmin() method
+def lf_ip(ip_df: pd.DataFrame) -> str:
+    return ip_df["Client IP address"].value_counts().idxmin()
+
+
+# Return events per seconds
+def e_ps(e_df: pd.DataFrame) -> float:
+    e_df["date"] = pd.to_datetime(
+        e_df["Timestamp in seconds since the epoch"], unit="ms"  # Convert unix time into datetime
+    )
+    min_unix_time_value = e_df["date"].value_counts().idxmin()  # Return first event
```

Comment:
dlaczaego kazda linia kodu zawiera inline comment, bardzo dziwnie to wyglada. Programista rozumie kod, wystraczy opis funkcji jako docstring - m.in. co funkcja robi według autora.